pipenv fails to skip extras

### Issue description

The reverse of https://github.com/pypa/pipenv/issues/4407: I want `extras` **not** to be installed

### Expected result

Install packages from `Pipfile.lock` whose `extras` does not match.

### Actual result

It attempts to install a package with `extras=systemd`

### Steps to replicate

You can check the `Pipfile` and `Pipfile.lock` from here:
- https://github.com/glensc/PlexTraktSync/tree/e1be343cbeba3e3ec2a2acf9354516a9dda15705
  - https://github.com/glensc/PlexTraktSync/blob/e1be343cbeba3e3ec2a2acf9354516a9dda15705/Pipfile
  - https://github.com/glensc/PlexTraktSync/blob/e1be343cbeba3e3ec2a2acf9354516a9dda15705/Pipfile.lock

-------------------------------------------------------------------------------
<details><summary>$ pipenv --support</summary>

Pipenv version: `'2023.11.15'`

Pipenv location: `'/*redacted*/python-3.11/lib/python3.11/site-packages/pipenv'`

Python location: `'/*redacted*/python-3.11/bin/python3.11'`

OS Name: `'posix'`

User pip version: `'23.3.1'`

user Python installations found:


PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.11.6',
 'os_name': 'posix',
 'platform_machine': 'arm64',
 'platform_python_implementation': 'CPython',
 'platform_release': '23.1.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT '
                     '2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000',
 'python_full_version': '3.11.6',
 'python_version': '3.11',
 'sys_platform': 'darwin'}
```

System environment variables:

  - *redacted*

Pipenv–specific environment variables:

 - `PIPENV_VERBOSITY`: `-1`

Debug–specific environment variables:

  - `PATH`: *redacted*
  - `SHELL`: `/bin/zsh`
  - `EDITOR`: `vim`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `*redacted*/PlexTraktSync`
  - `VIRTUAL_ENV`: `*redacted*/PlexTraktSync/.direnv/python-3.11`


---------------------------

Contents of `Pipfile` ('/*redacted*/PlexTraktSync/Pipfile'):

```toml
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]
attrs = "==23.1.0"
cattrs = {version="==23.2.3", python_version=">='3.7'"}
certifi = "==2023.11.17"
charset-normalizer = "==3.3.2"
click = "==8.1.7"
cysystemd = {version="==1.6.0", extras=["systemd"]}
deprecated = "==1.2.14"
idna = {version="==3.6", python_version=">='3.5'"}
inquirerpy = "==0.3.4"
markdown-it-py = {version="==3.0.0", python_version=">='3.7'"}
mdurl = {version="==0.1.2", python_version=">='3.7'"}
oauthlib = "==3.2.2"
pfzy = {version="==0.3.4", markers="python_version >= '3.7' and python_version < '4.0'"}
platformdirs = {version="==4.1.0", python_version=">='3.7'"}
plexapi = "==4.15.4"
prompt-toolkit = "==3.0.41"
pygments = "==2.17.2"
python-dotenv = "==1.0.0"
python-git-info = "==0.8.3"
pytimeparse = "==1.1.8"
pytrakt = "==3.4.24"
pyyaml = "==6.0.1"
requests = "==2.31.0"
requests-cache = "==1.1.0"
requests-oauthlib = {version="==1.3.1", python_version=">='3.4'"}
rich = "==13.7.0"
six = {version="==1.16.0", python_version=">='3.4'"}
tqdm = "==4.66.1"
url-normalize = {version="==1.4.3", python_version=">='3.6'"}
urllib3 = "==2.1.0"
wcwidth = "==0.2.12"
websocket-client = "==1.6.4"
wrapt = {version="==1.16.0", python_version=">='3.5'"}

[requires]
python_version = "3"

[scripts]
plextraktsync = "python -m plextraktsync"

```


Contents of `Pipfile.lock` ('/*redacted*/PlexTraktSync/Pipfile.lock'):

```json
{
    "_meta": {
        "hash": {
            "sha256": "9ab111f691b523913f6cc5a14e881bd60e254cf2425140f1cbf2709ff3a2d52d"
        },
        "pipfile-spec": 6,
        "requires": {
            "python_version": "3"
        },
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {
        "attrs": {
            "hashes": [
                "sha256:1f28b4522cdc2fb4256ac1a020c78acf9cba2c6b461ccd2c126f3aa8e8335d04",
                "sha256:6279836d581513a26f1bf235f9acd333bc9115683f14f7e8fae46c98fc50e015"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7'",
            "version": "==23.1.0"
        },
        "cattrs": {
            "hashes": [
                "sha256:0341994d94971052e9ee70662542699a3162ea1e0c62f7ce1b4a57f563685108",
                "sha256:a934090d95abaa9e911dac357e3a8699e0b4b14f8529bcc7d2b1ad9d51672b9f"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.8'",
            "version": "==23.2.3"
        },
        "certifi": {
            "hashes": [
                "sha256:9b469f3a900bf28dc19b8cfbf8019bf47f7fdd1a65a1d4ffb98fc14166beb4d1",
                "sha256:e036ab49d5b79556f99cfc2d9320b34cfbe5be05c5871b51de9329f0603b0474"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.6'",
            "version": "==2023.11.17"
        },
        "charset-normalizer": {
            "hashes": [
                "sha256:06435b539f889b1f6f4ac1758871aae42dc3a8c0e24ac9e60c2384973ad73027",
                "sha256:06a81e93cd441c56a9b65d8e1d043daeb97a3d0856d177d5c90ba85acb3db087",
                "sha256:0a55554a2fa0d408816b3b5cedf0045f4b8e1a6065aec45849de2d6f3f8e9786",
                "sha256:0b2b64d2bb6d3fb9112bafa732def486049e63de9618b5843bcdd081d8144cd8",
                "sha256:10955842570876604d404661fbccbc9c7e684caf432c09c715ec38fbae45ae09",
                "sha256:122c7fa62b130ed55f8f285bfd56d5f4b4a5b503609d181f9ad85e55c89f4185",
                "sha256:1ceae2f17a9c33cb48e3263960dc5fc8005351ee19db217e9b1bb15d28c02574",
                "sha256:1d3193f4a680c64b4b6a9115943538edb896edc190f0b222e73761716519268e",
                "sha256:1f79682fbe303db92bc2b1136016a38a42e835d932bab5b3b1bfcfbf0640e519",
                "sha256:2127566c664442652f024c837091890cb1942c30937add288223dc895793f898",
                "sha256:22afcb9f253dac0696b5a4be4a1c0f8762f8239e21b99680099abd9b2b1b2269",
                "sha256:25baf083bf6f6b341f4121c2f3c548875ee6f5339300e08be3f2b2ba1721cdd3",
                "sha256:2e81c7b9c8979ce92ed306c249d46894776a909505d8f5a4ba55b14206e3222f",
                "sha256:3287761bc4ee9e33561a7e058c72ac0938c4f57fe49a09eae428fd88aafe7bb6",
                "sha256:34d1c8da1e78d2e001f363791c98a272bb734000fcef47a491c1e3b0505657a8",
                "sha256:37e55c8e51c236f95b033f6fb391d7d7970ba5fe7ff453dad675e88cf303377a",
                "sha256:3d47fa203a7bd9c5b6cee4736ee84ca03b8ef23193c0d1ca99b5089f72645c73",
                "sha256:3e4d1f6587322d2788836a99c69062fbb091331ec940e02d12d179c1d53e25fc",
                "sha256:42cb296636fcc8b0644486d15c12376cb9fa75443e00fb25de0b8602e64c1714",
                "sha256:45485e01ff4d3630ec0d9617310448a8702f70e9c01906b0d0118bdf9d124cf2",
                "sha256:4a78b2b446bd7c934f5dcedc588903fb2f5eec172f3d29e52a9096a43722adfc",
                "sha256:4ab2fe47fae9e0f9dee8c04187ce5d09f48eabe611be8259444906793ab7cbce",
                "sha256:4d0d1650369165a14e14e1e47b372cfcb31d6ab44e6e33cb2d4e57265290044d",
                "sha256:549a3a73da901d5bc3ce8d24e0600d1fa85524c10287f6004fbab87672bf3e1e",
                "sha256:55086ee1064215781fff39a1af09518bc9255b50d6333f2e4c74ca09fac6a8f6",
                "sha256:572c3763a264ba47b3cf708a44ce965d98555f618ca42c926a9c1616d8f34269",
                "sha256:573f6eac48f4769d667c4442081b1794f52919e7edada77495aaed9236d13a96",
                "sha256:5b4c145409bef602a690e7cfad0a15a55c13320ff7a3ad7ca59c13bb8ba4d45d",
                "sha256:6463effa3186ea09411d50efc7d85360b38d5f09b870c48e4600f63af490e56a",
                "sha256:65f6f63034100ead094b8744b3b97965785388f308a64cf8d7c34f2f2e5be0c4",
                "sha256:663946639d296df6a2bb2aa51b60a2454ca1cb29835324c640dafb5ff2131a77",
                "sha256:6897af51655e3691ff853668779c7bad41579facacf5fd7253b0133308cf000d",
                "sha256:68d1f8a9e9e37c1223b656399be5d6b448dea850bed7d0f87a8311f1ff3dabb0",
                "sha256:6ac7ffc7ad6d040517be39eb591cac5ff87416c2537df6ba3cba3bae290c0fed",
                "sha256:6b3251890fff30ee142c44144871185dbe13b11bab478a88887a639655be1068",
                "sha256:6c4caeef8fa63d06bd437cd4bdcf3ffefe6738fb1b25951440d80dc7df8c03ac",
                "sha256:6ef1d82a3af9d3eecdba2321dc1b3c238245d890843e040e41e470ffa64c3e25",
                "sha256:753f10e867343b4511128c6ed8c82f7bec3bd026875576dfd88483c5c73b2fd8",
                "sha256:7cd13a2e3ddeed6913a65e66e94b51d80a041145a026c27e6bb76c31a853c6ab",
                "sha256:7ed9e526742851e8d5cc9e6cf41427dfc6068d4f5a3bb03659444b4cabf6bc26",
                "sha256:7f04c839ed0b6b98b1a7501a002144b76c18fb1c1850c8b98d458ac269e26ed2",
                "sha256:802fe99cca7457642125a8a88a084cef28ff0cf9407060f7b93dca5aa25480db",
                "sha256:80402cd6ee291dcb72644d6eac93785fe2c8b9cb30893c1af5b8fdd753b9d40f",
                "sha256:8465322196c8b4d7ab6d1e049e4c5cb460d0394da4a27d23cc242fbf0034b6b5",
                "sha256:86216b5cee4b06df986d214f664305142d9c76df9b6512be2738aa72a2048f99",
                "sha256:87d1351268731db79e0f8e745d92493ee2841c974128ef629dc518b937d9194c",
                "sha256:8bdb58ff7ba23002a4c5808d608e4e6c687175724f54a5dade5fa8c67b604e4d",
                "sha256:8c622a5fe39a48f78944a87d4fb8a53ee07344641b0562c540d840748571b811",
                "sha256:8d756e44e94489e49571086ef83b2bb8ce311e730092d2c34ca8f7d925cb20aa",
                "sha256:8f4a014bc36d3c57402e2977dada34f9c12300af536839dc38c0beab8878f38a",
                "sha256:9063e24fdb1e498ab71cb7419e24622516c4a04476b17a2dab57e8baa30d6e03",
                "sha256:90d558489962fd4918143277a773316e56c72da56ec7aa3dc3dbbe20fdfed15b",
                "sha256:923c0c831b7cfcb071580d3f46c4baf50f174be571576556269530f4bbd79d04",
                "sha256:95f2a5796329323b8f0512e09dbb7a1860c46a39da62ecb2324f116fa8fdc85c",
                "sha256:96b02a3dc4381e5494fad39be677abcb5e6634bf7b4fa83a6dd3112607547001",
                "sha256:9f96df6923e21816da7e0ad3fd47dd8f94b2a5ce594e00677c0013018b813458",
                "sha256:a10af20b82360ab00827f916a6058451b723b4e65030c5a18577c8b2de5b3389",
                "sha256:a50aebfa173e157099939b17f18600f72f84eed3049e743b68ad15bd69b6bf99",
                "sha256:a981a536974bbc7a512cf44ed14938cf01030a99e9b3a06dd59578882f06f985",
                "sha256:a9a8e9031d613fd2009c182b69c7b2c1ef8239a0efb1df3f7c8da66d5dd3d537",
                "sha256:ae5f4161f18c61806f411a13b0310bea87f987c7d2ecdbdaad0e94eb2e404238",
                "sha256:aed38f6e4fb3f5d6bf81bfa990a07806be9d83cf7bacef998ab1a9bd660a581f",
                "sha256:b01b88d45a6fcb69667cd6d2f7a9aeb4bf53760d7fc536bf679ec94fe9f3ff3d",
                "sha256:b261ccdec7821281dade748d088bb6e9b69e6d15b30652b74cbbac25e280b796",
                "sha256:b2b0a0c0517616b6869869f8c581d4eb2dd83a4d79e0ebcb7d373ef9956aeb0a",
                "sha256:b4a23f61ce87adf89be746c8a8974fe1c823c891d8f86eb218bb957c924bb143",
                "sha256:bd8f7df7d12c2db9fab40bdd87a7c09b1530128315d047a086fa3ae3435cb3a8",
                "sha256:beb58fe5cdb101e3a055192ac291b7a21e3b7ef4f67fa1d74e331a7f2124341c",
                "sha256:c002b4ffc0be611f0d9da932eb0f704fe2602a9a949d1f738e4c34c75b0863d5",
                "sha256:c083af607d2515612056a31f0a8d9e0fcb5876b7bfc0abad3ecd275bc4ebc2d5",
                "sha256:c180f51afb394e165eafe4ac2936a14bee3eb10debc9d9e4db8958fe36afe711",
                "sha256:c235ebd9baae02f1b77bcea61bce332cb4331dc3617d254df3323aa01ab47bd4",
                "sha256:cd70574b12bb8a4d2aaa0094515df2463cb429d8536cfb6c7ce983246983e5a6",
                "sha256:d0eccceffcb53201b5bfebb52600a5fb483a20b61da9dbc885f8b103cbe7598c",
                "sha256:d965bba47ddeec8cd560687584e88cf699fd28f192ceb452d1d7ee807c5597b7",
                "sha256:db364eca23f876da6f9e16c9da0df51aa4f104a972735574842618b8c6d999d4",
                "sha256:ddbb2551d7e0102e7252db79ba445cdab71b26640817ab1e3e3648dad515003b",
                "sha256:deb6be0ac38ece9ba87dea880e438f25ca3eddfac8b002a2ec3d9183a454e8ae",
                "sha256:e06ed3eb3218bc64786f7db41917d4e686cc4856944f53d5bdf83a6884432e12",
                "sha256:e27ad930a842b4c5eb8ac0016b0a54f5aebbe679340c26101df33424142c143c",
                "sha256:e537484df0d8f426ce2afb2d0f8e1c3d0b114b83f8850e5f2fbea0e797bd82ae",
                "sha256:eb00ed941194665c332bf8e078baf037d6c35d7c4f3102ea2d4f16ca94a26dc8",
                "sha256:eb6904c354526e758fda7167b33005998fb68c46fbc10e013ca97f21ca5c8887",
                "sha256:eb8821e09e916165e160797a6c17edda0679379a4be5c716c260e836e122f54b",
                "sha256:efcb3f6676480691518c177e3b465bcddf57cea040302f9f4e6e191af91174d4",
                "sha256:f27273b60488abe721a075bcca6d7f3964f9f6f067c8c4c605743023d7d3944f",
                "sha256:f30c3cb33b24454a82faecaf01b19c18562b1e89558fb6c56de4d9118a032fd5",
                "sha256:fb69256e180cb6c8a894fee62b3afebae785babc1ee98b81cdf68bbca1987f33",
                "sha256:fd1abc0d89e30cc4e02e4064dc67fcc51bd941eb395c502aac3ec19fab46b519",
                "sha256:ff8fa367d09b717b2a17a052544193ad76cd49979c805768879cb63d9ca50561"
            ],
            "index": "pypi",
            "markers": "python_full_version >= '3.7.0'",
            "version": "==3.3.2"
        },
        "click": {
            "hashes": [
                "sha256:ae74fb96c20a0277a1d615f1e4d73c8414f5a98db8b799a7931d1582f3390c28",
                "sha256:ca9853ad459e787e2192211578cc907e7594e294c7ccc834310722b41b9ca6de"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7'",
            "version": "==8.1.7"
        },
        "cysystemd": {
            "extras": [
                "systemd"
            ],
            "hashes": [
                "sha256:5224dd8fee146de08528bbf685edb177568246c7728bbb548b2257c9a44a2454"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7' and python_version < '4'",
            "version": "==1.6.0"
        },
        "deprecated": {
            "hashes": [
                "sha256:6fac8b097794a90302bdbb17b9b815e732d3c4720583ff1b198499d78470466c",
                "sha256:e5323eb936458dccc2582dc6f9c322c852a775a27065ff2b0c4970b9d53d01b3"
            ],
            "index": "pypi",
            "markers": "python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
            "version": "==1.2.14"
        },
        "idna": {
            "hashes": [
                "sha256:9ecdbbd083b06798ae1e86adcbfe8ab1479cf864e4ee30fe4e46a003d12491ca",
                "sha256:c05567e9c24a6b9faaa835c4821bad0590fbb9d5779e7caa6e1cc4978e7eb24f"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.5'",
            "version": "==3.6"
        },
        "inquirerpy": {
            "hashes": [
                "sha256:89d2ada0111f337483cb41ae31073108b2ec1e618a49d7110b0d7ade89fc197e",
                "sha256:c65fdfbac1fa00e3ee4fb10679f4d3ed7a012abf4833910e63c295827fe2a7d4"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7' and python_version < '4.0'",
            "version": "==0.3.4"
        },
        "markdown-it-py": {
            "hashes": [
                "sha256:355216845c60bd96232cd8d8c40e8f9765cc86f46880e43a8fd22dc1a1a8cab1",
                "sha256:e3f60a94fa066dc52ec76661e37c851cb232d92f9886b15cb560aaada2df8feb"
            ],
            "markers": "python_version >= '3.7' and python_version >= '3.8'",
            "version": "==3.0.0"
        },
        "mdurl": {
            "hashes": [
                "sha256:84008a41e51615a49fc9966191ff91509e3c40b939176e643fd50a5c2196b8f8",
                "sha256:bb413d29f5eea38f31dd4754dd7377d4465116fb207585f97bf925588687c1ba"
            ],
            "markers": "python_version >= '3.7'",
            "version": "==0.1.2"
        },
        "oauthlib": {
            "hashes": [
                "sha256:8139f29aac13e25d502680e9e19963e83f16838d48a0d71c287fe40e7067fbca",
                "sha256:9859c40929662bec5d64f34d01c99e093149682a3f38915dc0655d5a633dd918"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.6'",
            "version": "==3.2.2"
        },
        "pfzy": {
            "hashes": [
                "sha256:5f50d5b2b3207fa72e7ec0ef08372ef652685470974a107d0d4999fc5a903a96",
                "sha256:717ea765dd10b63618e7298b2d98efd819e0b30cd5905c9707223dceeb94b3f1"
            ],
            "markers": "python_version >= '3.7' and python_version < '4.0'",
            "version": "==0.3.4"
        },
        "platformdirs": {
            "hashes": [
                "sha256:11c8f37bcca40db96d8144522d925583bdb7a31f7b0e37e3ed4318400a8e2380",
                "sha256:906d548203468492d432bcb294d4bc2fff751bf84971fbb2c10918cc206ee420"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.8'",
            "version": "==4.1.0"
        },
        "plexapi": {
            "hashes": [
                "sha256:26b774e2934fd0ccb0e07d31ee56d7df10e1133e74fe50160a85ef527ee6ff3c",
                "sha256:9655f91216af0c6ff6cffbda293183ec3a7a10b68c5e8e3294217a6b95f33cd4"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.8'",
            "version": "==4.15.4"
        },
        "prompt-toolkit": {
            "hashes": [
                "sha256:941367d97fc815548822aa26c2a269fdc4eb21e9ec05fc5d447cf09bad5d75f0",
                "sha256:f36fe301fafb7470e86aaf90f036eef600a3210be4decf461a5b1ca8403d3cb2"
            ],
            "index": "pypi",
            "markers": "python_full_version >= '3.7.0'",
            "version": "==3.0.41"
        },
        "pygments": {
            "hashes": [
                "sha256:b27c2826c47d0f3219f29554824c30c5e8945175d888647acd804ddd04af846c",
                "sha256:da46cec9fd2de5be3a8a784f434e4c4ab670b4ff54d605c4c2717e9d49c4c367"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7'",
            "version": "==2.17.2"
        },
        "python-dotenv": {
            "hashes": [
                "sha256:a8df96034aae6d2d50a4ebe8216326c61c3eb64836776504fcca410e5937a3ba",
                "sha256:f5971a9226b701070a4bf2c38c89e5a3f0d64de8debda981d1db98583009122a"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.8'",
            "version": "==1.0.0"
        },
        "python-git-info": {
            "hashes": [
                "sha256:c08a2764a16da029e4db909ab8436ae484f8846405f83b1bcfd415fe5e3ab30d",
                "sha256:f8911aa590170434272c834d1967b74faea71ee573e50cbe0b88e938e3db406b"
            ],
            "index": "pypi",
            "version": "==0.8.3"
        },
        "pytimeparse": {
            "hashes": [
                "sha256:04b7be6cc8bd9f5647a6325444926c3ac34ee6bc7e69da4367ba282f076036bd",
                "sha256:e86136477be924d7e670646a98561957e8ca7308d44841e21f5ddea757556a0a"
            ],
            "index": "pypi",
            "version": "==1.1.8"
        },
        "pytrakt": {
            "hashes": [
                "sha256:1239e068a010ab6a7062d772b11f30573368a1eb8d36cb7d50888bee311ec50d",
                "sha256:6bb9ba14cc7edb340259be18edbd13b00ab2741aa29b0c5c203d736e725d4651"
            ],
            "index": "pypi",
            "version": "==3.4.24"
        },
        "pyyaml": {
            "hashes": [
                "sha256:04ac92ad1925b2cff1db0cfebffb6ffc43457495c9b3c39d3fcae417d7125dc5",
                "sha256:062582fca9fabdd2c8b54a3ef1c978d786e0f6b3a1510e0ac93ef59e0ddae2bc",
                "sha256:0d3304d8c0adc42be59c5f8a4d9e3d7379e6955ad754aa9d6ab7a398b59dd1df",
                "sha256:1635fd110e8d85d55237ab316b5b011de701ea0f29d07611174a1b42f1444741",
                "sha256:184c5108a2aca3c5b3d3bf9395d50893a7ab82a38004c8f61c258d4428e80206",
                "sha256:18aeb1bf9a78867dc38b259769503436b7c72f7a1f1f4c93ff9a17de54319b27",
                "sha256:1d4c7e777c441b20e32f52bd377e0c409713e8bb1386e1099c2415f26e479595",
                "sha256:1e2722cc9fbb45d9b87631ac70924c11d3a401b2d7f410cc0e3bbf249f2dca62",
                "sha256:1fe35611261b29bd1de0070f0b2f47cb6ff71fa6595c077e42bd0c419fa27b98",
                "sha256:28c119d996beec18c05208a8bd78cbe4007878c6dd15091efb73a30e90539696",
                "sha256:326c013efe8048858a6d312ddd31d56e468118ad4cdeda36c719bf5bb6192290",
                "sha256:40df9b996c2b73138957fe23a16a4f0ba614f4c0efce1e9406a184b6d07fa3a9",
                "sha256:42f8152b8dbc4fe7d96729ec2b99c7097d656dc1213a3229ca5383f973a5ed6d",
                "sha256:49a183be227561de579b4a36efbb21b3eab9651dd81b1858589f796549873dd6",
                "sha256:4fb147e7a67ef577a588a0e2c17b6db51dda102c71de36f8549b6816a96e1867",
                "sha256:50550eb667afee136e9a77d6dc71ae76a44df8b3e51e41b77f6de2932bfe0f47",
                "sha256:510c9deebc5c0225e8c96813043e62b680ba2f9c50a08d3724c7f28a747d1486",
                "sha256:5773183b6446b2c99bb77e77595dd486303b4faab2b086e7b17bc6bef28865f6",
                "sha256:596106435fa6ad000c2991a98fa58eeb8656ef2325d7e158344fb33864ed87e3",
                "sha256:6965a7bc3cf88e5a1c3bd2e0b5c22f8d677dc88a455344035f03399034eb3007",
                "sha256:69b023b2b4daa7548bcfbd4aa3da05b3a74b772db9e23b982788168117739938",
                "sha256:6c22bec3fbe2524cde73d7ada88f6566758a8f7227bfbf93a408a9d86bcc12a0",
                "sha256:704219a11b772aea0d8ecd7058d0082713c3562b4e271b849ad7dc4a5c90c13c",
                "sha256:7e07cbde391ba96ab58e532ff4803f79c4129397514e1413a7dc761ccd755735",
                "sha256:81e0b275a9ecc9c0c0c07b4b90ba548307583c125f54d5b6946cfee6360c733d",
                "sha256:855fb52b0dc35af121542a76b9a84f8d1cd886ea97c84703eaa6d88e37a2ad28",
                "sha256:8d4e9c88387b0f5c7d5f281e55304de64cf7f9c0021a3525bd3b1c542da3b0e4",
                "sha256:9046c58c4395dff28dd494285c82ba00b546adfc7ef001486fbf0324bc174fba",
                "sha256:9eb6caa9a297fc2c2fb8862bc5370d0303ddba53ba97e71f08023b6cd73d16a8",
                "sha256:a0cd17c15d3bb3fa06978b4e8958dcdc6e0174ccea823003a106c7d4d7899ac5",
                "sha256:afd7e57eddb1a54f0f1a974bc4391af8bcce0b444685d936840f125cf046d5bd",
                "sha256:b1275ad35a5d18c62a7220633c913e1b42d44b46ee12554e5fd39c70a243d6a3",
                "sha256:b786eecbdf8499b9ca1d697215862083bd6d2a99965554781d0d8d1ad31e13a0",
                "sha256:ba336e390cd8e4d1739f42dfe9bb83a3cc2e80f567d8805e11b46f4a943f5515",
                "sha256:baa90d3f661d43131ca170712d903e6295d1f7a0f595074f151c0aed377c9b9c",
                "sha256:bc1bf2925a1ecd43da378f4db9e4f799775d6367bdb94671027b73b393a7c42c",
                "sha256:bd4af7373a854424dabd882decdc5579653d7868b8fb26dc7d0e99f823aa5924",
                "sha256:bf07ee2fef7014951eeb99f56f39c9bb4af143d8aa3c21b1677805985307da34",
                "sha256:bfdf460b1736c775f2ba9f6a92bca30bc2095067b8a9d77876d1fad6cc3b4a43",
                "sha256:c8098ddcc2a85b61647b2590f825f3db38891662cfc2fc776415143f599bb859",
                "sha256:d2b04aac4d386b172d5b9692e2d2da8de7bfb6c387fa4f801fbf6fb2e6ba4673",
                "sha256:d483d2cdf104e7c9fa60c544d92981f12ad66a457afae824d146093b8c294c54",
                "sha256:d858aa552c999bc8a8d57426ed01e40bef403cd8ccdd0fc5f6f04a00414cac2a",
                "sha256:e7d73685e87afe9f3b36c799222440d6cf362062f78be1013661b00c5c6f678b",
                "sha256:f003ed9ad21d6a4713f0a9b5a7a0a79e08dd0f221aff4525a2be4c346ee60aab",
                "sha256:f22ac1c3cac4dbc50079e965eba2c1058622631e526bd9afd45fedd49ba781fa",
                "sha256:faca3bdcf85b2fc05d06ff3fbc1f83e1391b3e724afa3feba7d13eeab355484c",
                "sha256:fca0e3a251908a499833aa292323f32437106001d436eca0e6e7833256674585",
                "sha256:fd1592b3fdf65fff2ad0004b5e363300ef59ced41c2e6b3a99d4089fa8c5435d",
                "sha256:fd66fc5d0da6d9815ba2cebeb4205f95818ff4b79c3ebe268e75d961704af52f"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.6'",
            "version": "==6.0.1"
        },
        "requests": {
            "hashes": [
                "sha256:58cd2187c01e70e6e26505bca751777aa9f2ee0b7f4300988b709f44e013003f",
                "sha256:942c5a758f98d790eaed1a29cb6eefc7ffb0d1cf7af05c3d2791656dbd6ad1e1"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7'",
            "version": "==2.31.0"
        },
        "requests-cache": {
            "hashes": [
                "sha256:178282bce704b912c59e7f88f367c42bddd6cde6bf511b2a3e3cfb7e5332a92a",
                "sha256:41b79166aa8e300cc4de982f7ab7c52af914a785160be1eda25c6e9265969a67"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7' and python_version < '4.0'",
            "version": "==1.1.0"
        },
        "requests-oauthlib": {
            "hashes": [
                "sha256:2577c501a2fb8d05a304c09d090d6e47c306fef15809d102b327cf8364bddab5",
                "sha256:75beac4a47881eeb94d5ea5d6ad31ef88856affe2332b9aafb52c6452ccf0d7a"
            ],
            "markers": "python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3' and python_version >= '3.4'",
            "version": "==1.3.1"
        },
        "rich": {
            "hashes": [
                "sha256:5cb5123b5cf9ee70584244246816e9114227e0b98ad9176eede6ad54bf5403fa",
                "sha256:6da14c108c4866ee9520bbffa71f6fe3962e193b7da68720583850cd4548e235"
            ],
            "index": "pypi",
            "markers": "python_full_version >= '3.7.0'",
            "version": "==13.7.0"
        },
        "six": {
            "hashes": [
                "sha256:1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926",
                "sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254"
            ],
            "markers": "python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3' and python_version >= '3.4'",
            "version": "==1.16.0"
        },
        "tqdm": {
            "hashes": [
                "sha256:d302b3c5b53d47bce91fea46679d9c3c6508cf6332229aa1e7d8653723793386",
                "sha256:d88e651f9db8d8551a62556d3cff9e3034274ca5d66e93197cf2490e2dcb69c7"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.7'",
            "version": "==4.66.1"
        },
        "url-normalize": {
            "hashes": [
                "sha256:d23d3a070ac52a67b83a1c59a0e68f8608d1cd538783b401bc9de2c0fac999b2",
                "sha256:ec3c301f04e5bb676d333a7fa162fa977ad2ca04b7e652bfc9fac4e405728eed"
            ],
            "markers": "python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5' and python_version >= '3.6'",
            "version": "==1.4.3"
        },
        "urllib3": {
            "hashes": [
                "sha256:55901e917a5896a349ff771be919f8bd99aff50b79fe58fec595eb37bbc56bb3",
                "sha256:df7aa8afb0148fa78488e7899b2c59b5f4ffcfa82e6c54ccb9dd37c1d7b52d54"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.8'",
            "version": "==2.1.0"
        },
        "wcwidth": {
            "hashes": [
                "sha256:f01c104efdf57971bcb756f054dd58ddec5204dd15fa31d6503ea57947d97c02",
                "sha256:f26ec43d96c8cbfed76a5075dac87680124fa84e0855195a6184da9c187f133c"
            ],
            "index": "pypi",
            "version": "==0.2.12"
        },
        "websocket-client": {
            "hashes": [
                "sha256:084072e0a7f5f347ef2ac3d8698a5e0b4ffbfcab607628cadabc650fc9a83a24",
                "sha256:b3324019b3c28572086c4a319f91d1dcd44e6e11cd340232978c684a7650d0df"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.8'",
            "version": "==1.6.4"
        },
        "wrapt": {
            "hashes": [
                "sha256:0d2691979e93d06a95a26257adb7bfd0c93818e89b1406f5a28f36e0d8c1e1fc",
                "sha256:14d7dc606219cdd7405133c713f2c218d4252f2a469003f8c46bb92d5d095d81",
                "sha256:1a5db485fe2de4403f13fafdc231b0dbae5eca4359232d2efc79025527375b09",
                "sha256:1acd723ee2a8826f3d53910255643e33673e1d11db84ce5880675954183ec47e",
                "sha256:1ca9b6085e4f866bd584fb135a041bfc32cab916e69f714a7d1d397f8c4891ca",
                "sha256:1dd50a2696ff89f57bd8847647a1c363b687d3d796dc30d4dd4a9d1689a706f0",
                "sha256:2076fad65c6736184e77d7d4729b63a6d1ae0b70da4868adeec40989858eb3fb",
                "sha256:2a88e6010048489cda82b1326889ec075a8c856c2e6a256072b28eaee3ccf487",
                "sha256:3ebf019be5c09d400cf7b024aa52b1f3aeebeff51550d007e92c3c1c4afc2a40",
                "sha256:418abb18146475c310d7a6dc71143d6f7adec5b004ac9ce08dc7a34e2babdc5c",
                "sha256:43aa59eadec7890d9958748db829df269f0368521ba6dc68cc172d5d03ed8060",
                "sha256:44a2754372e32ab315734c6c73b24351d06e77ffff6ae27d2ecf14cf3d229202",
                "sha256:490b0ee15c1a55be9c1bd8609b8cecd60e325f0575fc98f50058eae366e01f41",
                "sha256:49aac49dc4782cb04f58986e81ea0b4768e4ff197b57324dcbd7699c5dfb40b9",
                "sha256:5eb404d89131ec9b4f748fa5cfb5346802e5ee8836f57d516576e61f304f3b7b",
                "sha256:5f15814a33e42b04e3de432e573aa557f9f0f56458745c2074952f564c50e664",
                "sha256:5f370f952971e7d17c7d1ead40e49f32345a7f7a5373571ef44d800d06b1899d",
                "sha256:66027d667efe95cc4fa945af59f92c5a02c6f5bb6012bff9e60542c74c75c362",
                "sha256:66dfbaa7cfa3eb707bbfcd46dab2bc6207b005cbc9caa2199bcbc81d95071a00",
                "sha256:685f568fa5e627e93f3b52fda002c7ed2fa1800b50ce51f6ed1d572d8ab3e7fc",
                "sha256:6906c4100a8fcbf2fa735f6059214bb13b97f75b1a61777fcf6432121ef12ef1",
                "sha256:6a42cd0cfa8ffc1915aef79cb4284f6383d8a3e9dcca70c445dcfdd639d51267",
                "sha256:6dcfcffe73710be01d90cae08c3e548d90932d37b39ef83969ae135d36ef3956",
                "sha256:6f6eac2360f2d543cc875a0e5efd413b6cbd483cb3ad7ebf888884a6e0d2e966",
                "sha256:72554a23c78a8e7aa02abbd699d129eead8b147a23c56e08d08dfc29cfdddca1",
                "sha256:73870c364c11f03ed072dda68ff7aea6d2a3a5c3fe250d917a429c7432e15228",
                "sha256:73aa7d98215d39b8455f103de64391cb79dfcad601701a3aa0dddacf74911d72",
                "sha256:75ea7d0ee2a15733684badb16de6794894ed9c55aa5e9903260922f0482e687d",
                "sha256:7bd2d7ff69a2cac767fbf7a2b206add2e9a210e57947dd7ce03e25d03d2de292",
                "sha256:807cc8543a477ab7422f1120a217054f958a66ef7314f76dd9e77d3f02cdccd0",
                "sha256:8e9723528b9f787dc59168369e42ae1c3b0d3fadb2f1a71de14531d321ee05b0",
                "sha256:9090c9e676d5236a6948330e83cb89969f433b1943a558968f659ead07cb3b36",
                "sha256:9153ed35fc5e4fa3b2fe97bddaa7cbec0ed22412b85bcdaf54aeba92ea37428c",
                "sha256:9159485323798c8dc530a224bd3ffcf76659319ccc7bbd52e01e73bd0241a0c5",
                "sha256:941988b89b4fd6b41c3f0bfb20e92bd23746579736b7343283297c4c8cbae68f",
                "sha256:94265b00870aa407bd0cbcfd536f17ecde43b94fb8d228560a1e9d3041462d73",
                "sha256:98b5e1f498a8ca1858a1cdbffb023bfd954da4e3fa2c0cb5853d40014557248b",
                "sha256:9b201ae332c3637a42f02d1045e1d0cccfdc41f1f2f801dafbaa7e9b4797bfc2",
                "sha256:a0ea261ce52b5952bf669684a251a66df239ec6d441ccb59ec7afa882265d593",
                "sha256:a33a747400b94b6d6b8a165e4480264a64a78c8a4c734b62136062e9a248dd39",
                "sha256:a452f9ca3e3267cd4d0fcf2edd0d035b1934ac2bd7e0e57ac91ad6b95c0c6389",
                "sha256:a86373cf37cd7764f2201b76496aba58a52e76dedfaa698ef9e9688bfd9e41cf",
                "sha256:ac83a914ebaf589b69f7d0a1277602ff494e21f4c2f743313414378f8f50a4cf",
                "sha256:aefbc4cb0a54f91af643660a0a150ce2c090d3652cf4052a5397fb2de549cd89",
                "sha256:b3646eefa23daeba62643a58aac816945cadc0afaf21800a1421eeba5f6cfb9c",
                "sha256:b47cfad9e9bbbed2339081f4e346c93ecd7ab504299403320bf85f7f85c7d46c",
                "sha256:b935ae30c6e7400022b50f8d359c03ed233d45b725cfdd299462f41ee5ffba6f",
                "sha256:bb2dee3874a500de01c93d5c71415fcaef1d858370d405824783e7a8ef5db440",
                "sha256:bc57efac2da352a51cc4658878a68d2b1b67dbe9d33c36cb826ca449d80a8465",
                "sha256:bf5703fdeb350e36885f2875d853ce13172ae281c56e509f4e6eca049bdfb136",
                "sha256:c31f72b1b6624c9d863fc095da460802f43a7c6868c5dda140f51da24fd47d7b",
                "sha256:c5cd603b575ebceca7da5a3a251e69561bec509e0b46e4993e1cac402b7247b8",
                "sha256:d2efee35b4b0a347e0d99d28e884dfd82797852d62fcd7ebdeee26f3ceb72cf3",
                "sha256:d462f28826f4657968ae51d2181a074dfe03c200d6131690b7d65d55b0f360f8",
                "sha256:d5e49454f19ef621089e204f862388d29e6e8d8b162efce05208913dde5b9ad6",
                "sha256:da4813f751142436b075ed7aa012a8778aa43a99f7b36afe9b742d3ed8bdc95e",
                "sha256:db2e408d983b0e61e238cf579c09ef7020560441906ca990fe8412153e3b291f",
                "sha256:db98ad84a55eb09b3c32a96c576476777e87c520a34e2519d3e59c44710c002c",
                "sha256:dbed418ba5c3dce92619656802cc5355cb679e58d0d89b50f116e4a9d5a9603e",
                "sha256:dcdba5c86e368442528f7060039eda390cc4091bfd1dca41e8046af7c910dda8",
                "sha256:decbfa2f618fa8ed81c95ee18a387ff973143c656ef800c9f24fb7e9c16054e2",
                "sha256:e4fdb9275308292e880dcbeb12546df7f3e0f96c6b41197e0cf37d2826359020",
                "sha256:eb1b046be06b0fce7249f1d025cd359b4b80fc1c3e24ad9eca33e0dcdb2e4a35",
                "sha256:eb6e651000a19c96f452c85132811d25e9264d836951022d6e81df2fff38337d",
                "sha256:ed867c42c268f876097248e05b6117a65bcd1e63b779e916fe2e33cd6fd0d3c3",
                "sha256:edfad1d29c73f9b863ebe7082ae9321374ccb10879eeabc84ba3b69f2579d537",
                "sha256:f2058f813d4f2b5e3a9eb2eb3faf8f1d99b81c3e51aeda4b168406443e8ba809",
                "sha256:f6b2d0c6703c988d334f297aa5df18c45e97b0af3679bb75059e0e0bd8b1069d",
                "sha256:f8212564d49c50eb4565e502814f694e240c55551a5f1bc841d4fcaabb0a9b8a",
                "sha256:ffa565331890b90056c01db69c0fe634a776f8019c143a5ae265f9c6bc4bd6d4"
            ],
            "index": "pypi",
            "markers": "python_version >= '3.6'",
            "version": "==1.16.0"
        }
    },
    "develop": {}
}

```
</details>

Install log

<details>

```
➜ pipenv install
Loading .env environment variables...
Installing dependencies from Pipfile.lock (a2d52d)...
[pipenv.exceptions.InstallError]: Collecting cysystemd==1.6.0 (from cysystemd[systemd]==1.6.0->-r /var/folders/f1/qj3m1q9507d2_7rnqwfgzyrm0000gp/T/pipenv-p_70yxsa-requirements/pipenv-hmocb6fe-hashed-reqs.txt (line 1))
[pipenv.exceptions.InstallError]:   Using cached cysystemd-1.6.0.tar.gz (296 kB)
[pipenv.exceptions.InstallError]:   Installing build dependencies: started
[pipenv.exceptions.InstallError]:   Installing build dependencies: finished with status 'done'
[pipenv.exceptions.InstallError]:   Getting requirements to build wheel: started
[pipenv.exceptions.InstallError]:   Getting requirements to build wheel: finished with status 'done'
[pipenv.exceptions.InstallError]:   Preparing metadata (pyproject.toml): started
[pipenv.exceptions.InstallError]:   Preparing metadata (pyproject.toml): finished with status 'done'
[pipenv.exceptions.InstallError]: Building wheels for collected packages: cysystemd
[pipenv.exceptions.InstallError]:   Building wheel for cysystemd (pyproject.toml): started
[pipenv.exceptions.InstallError]:   Building wheel for cysystemd (pyproject.toml): finished with status 'error'
[pipenv.exceptions.InstallError]: Failed to build cysystemd
[pipenv.exceptions.InstallError]: error: subprocess-exited-with-error
[pipenv.exceptions.InstallError]:   
[pipenv.exceptions.InstallError]:   × Building wheel for cysystemd (pyproject.toml) did not run successfully.
[pipenv.exceptions.InstallError]:   │ exit code: 1
[pipenv.exceptions.InstallError]:   ╰─> [25 lines of output]
[pipenv.exceptions.InstallError]:       /private/var/folders/f1/qj3m1q9507d2_7rnqwfgzyrm0000gp/T/pip-build-env-st96c0ja/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py:265: UserWarning: Unknown distribution option: 'build_requires'
[pipenv.exceptions.InstallError]:         warnings.warn(msg)
[pipenv.exceptions.InstallError]:       running bdist_wheel
[pipenv.exceptions.InstallError]:       running build
[pipenv.exceptions.InstallError]:       running build_py
[pipenv.exceptions.InstallError]:       creating build
[pipenv.exceptions.InstallError]:       creating build/lib.macosx-13-arm64-cpython-311
[pipenv.exceptions.InstallError]:       creating build/lib.macosx-13-arm64-cpython-311/cysystemd
[pipenv.exceptions.InstallError]:       copying cysystemd/journal.py -> build/lib.macosx-13-arm64-cpython-311/cysystemd
[pipenv.exceptions.InstallError]:       copying cysystemd/__init__.py -> build/lib.macosx-13-arm64-cpython-311/cysystemd
[pipenv.exceptions.InstallError]:       copying cysystemd/async_reader.py -> build/lib.macosx-13-arm64-cpython-311/cysystemd
[pipenv.exceptions.InstallError]:       copying cysystemd/daemon.py -> build/lib.macosx-13-arm64-cpython-311/cysystemd
[pipenv.exceptions.InstallError]:       copying cysystemd/py.typed -> build/lib.macosx-13-arm64-cpython-311/cysystemd
[pipenv.exceptions.InstallError]:       warning: build_py: byte-compiling is disabled, skipping.
[pipenv.exceptions.InstallError]:       
[pipenv.exceptions.InstallError]:       running build_ext
[pipenv.exceptions.InstallError]:       building 'cysystemd._daemon' extension
[pipenv.exceptions.InstallError]:       creating build/temp.macosx-13-arm64-cpython-311
[pipenv.exceptions.InstallError]:       creating build/temp.macosx-13-arm64-cpython-311/cysystemd
[pipenv.exceptions.InstallError]:       clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -I/Users/glen/Library/Caches/glen/direnv/PlexTraktSync/python-3.11/include -I/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c cysystemd/_daemon.c -o build/temp.macosx-13-arm64-cpython-311/cysystemd/_daemon.o
[pipenv.exceptions.InstallError]:       cysystemd/_daemon.c:1202:10: fatal error: 'systemd/sd-daemon.h' file not found
[pipenv.exceptions.InstallError]:       #include <systemd/sd-daemon.h>
[pipenv.exceptions.InstallError]:                ^~~~~~~~~~~~~~~~~~~~~
[pipenv.exceptions.InstallError]:       1 error generated.
[pipenv.exceptions.InstallError]:       error: command '/usr/bin/clang' failed with exit code 1
[pipenv.exceptions.InstallError]:       [end of output]
[pipenv.exceptions.InstallError]:   
[pipenv.exceptions.InstallError]:   note: This error originates from a subprocess, and is likely not a problem with pip.
[pipenv.exceptions.InstallError]:   ERROR: Failed building wheel for cysystemd
[pipenv.exceptions.InstallError]: ERROR: Could not build wheels for cysystemd, which is required to install pyproject.toml-based projects
ERROR: Couldn't install package: {}
 Package installation failed...
```

</details>

--- a/pipenv/cli/options.py
+++ b/pipenv/cli/options.py
@@ -184,6 +184,29 @@ def categories_option(f):
     )(f)
 
 
+def extras_option(f):
+    def callback(ctx, param, value):
+        state = ctx.ensure_object(State)
+        if value:
+            extras = parse_categories(value)
+            # --extras always includes 'packages' so users get the defaults
+            # plus the requested optional categories.
+            if "packages" not in state.installstate.categories:
+                state.installstate.categories.insert(0, "packages")
+            state.installstate.categories += extras
+        return value
+
+    return option(
+        "--extras",
+        nargs=1,
+        required=False,
+        callback=callback,
+        expose_value=False,
+        type=click_types.STRING,
+        help="Install optional extra categories alongside default packages.",
+    )(f)
+
+
 def all_categories_option(f):
     def callback(ctx, param, value):
         state = ctx.ensure_object(State)
@@ -613,6 +636,7 @@ def install_base_options(f):
 def uninstall_options(f):
     f = install_base_options(f)
     f = categories_option(f)
+    f = extras_option(f)
     f = uninstall_dev_option(f)
     f = editable_option(f)
     f = package_arg(f)
@@ -632,6 +656,7 @@ def sync_options(f):
     f = install_base_options(f)
     f = install_dev_option(f)
     f = categories_option(f)
+    f = extras_option(f)
     f = all_categories_option(f)
     return f
 

--- a/tests/unit/test_core.py
+++ b/tests/unit/test_core.py
@@ -767,3 +767,85 @@ def test_install_build_system_packages_calls_pip_install(project):
     call_kwargs = mock_pip_install.call_args
     assert call_kwargs[1]["deps"] == build_requires
     assert call_kwargs[1]["ignore_hashes"] is True
+
+
+
+# --- Tests for --extras CLI option ---
+
+
+@pytest.mark.core
+def test_parse_extras_single():
+    """Test that extras_option parses a single extra category."""
+    from pipenv.cli.options import parse_categories
+
+    result = parse_categories("systemd")
+    assert result == ["systemd"]
+
+
+@pytest.mark.core
+def test_parse_extras_multiple_comma():
+    """Test that extras_option parses comma-separated extras."""
+    from pipenv.cli.options import parse_categories
+
+    result = parse_categories("systemd,monitoring")
+    assert result == ["systemd", "monitoring"]
+
+
+@pytest.mark.core
+def test_parse_extras_multiple_space():
+    """Test that extras_option parses space-separated extras."""
+    from pipenv.cli.options import parse_categories
+
+    result = parse_categories("systemd monitoring")
+    assert result == ["systemd", "monitoring"]
+
+
+@pytest.mark.core
+def test_extras_option_adds_packages_category():
+    """Test that --extras ensures 'packages' is in the categories list."""
+    from pipenv.cli.options import InstallState
+
+    state = InstallState()
+    assert state.categories == []
+
+    # Simulate what extras_option callback does
+    extras = ["systemd"]
+    if "packages" not in state.categories:
+        state.categories.insert(0, "packages")
+    state.categories += extras
+
+    assert state.categories == ["packages", "systemd"]
+
+
+@pytest.mark.core
+def test_extras_option_does_not_duplicate_packages():
+    """Test that --extras doesn't duplicate 'packages' if already present."""
+    from pipenv.cli.options import InstallState
+
+    state = InstallState()
+    state.categories = ["packages"]
+
+    # Simulate what extras_option callback does
+    extras = ["systemd"]
+    if "packages" not in state.categories:
+        state.categories.insert(0, "packages")
+    state.categories += extras
+
+    assert state.categories == ["packages", "systemd"]
+
+
+@pytest.mark.core
+def test_extras_with_dev_categories():
+    """Test that --extras works alongside --dev categories."""
+    from pipenv.cli.options import InstallState
+
+    state = InstallState()
+    state.categories = ["dev-packages"]  # Simulates --dev being set first
+
+    # Simulate what extras_option callback does
+    extras = ["systemd"]
+    if "packages" not in state.categories:
+        state.categories.insert(0, "packages")
+    state.categories += extras
+
+    assert state.categories == ["packages", "dev-packages", "systemd"]

